contributed_definitions/nyaml/SPM/NXstm.yaml

```diff
@@ -0,0 +1,780 @@
+category: application
+doc: |
+  An application definition to describe Scanning Tunneling Microscopy (STM).
+
+# https://thesis.library.caltech.edu/1943/4/03chapter3.pdf
+# TODO: Replace rename NXenvironment to experiment_environment in NXstm, NXspm and NXafm
+type: group
+NXstm(NXspm):
+  (NXentry):
+    definition:
+      doc: |
+        Name of the definition that is used for the application.
+      enumeration: [NXstm]
+    scan_mode:
+      doc: |
+        The that is performed. Sometimes under the
+        same experiment setup, both constant height and current are measured.
+      enumeration: [constant height, constant current]
+    experiment_instrument(NXinstrument):
+      doc: |
+        The group explains the instrumentation of the STM experiment such
+        as current sensor, lock-in amplifier etc.
+      lockin_amplifier(NXlockin):
+        doc: |
+          The lock-in amplifier information. The device is being used to extract
+          the very weak signal buried in noisy signals.
+        modulation_signal_type:
+          exists: optional
+          unit: NX_UNITLESS
+          doc: |
+            This is the signal on which the modulation voltage or current will
+            be added.
+        modulation_frequency(NX_NUMBER):
+          exists: optional
+          doc: |
+            The frequency of the sine modulation that is used to modulate the signal in
+            lock-in.
+      scan_environment(NXenvironment):
+        doc: |
+          The environment information for stm or sts experiment.
+        (NXscan_control):
+          doc: |
+            The scan control information like scan region or phase space, type of scan
+            (e.g. mesh, spiral, etc.), and scan speed, etc.
+          scan_type:
+            doc: |
+              The type of scan like mesh, spiral, etc. For STS experiment, the scan type is
+              usually a single-point scan (trajectory scan).
+            enumeration: [mesh]
```

Comment:
why did you remove other types of scan available in NXscan_control/scan_type? I can imagine somebody using trajectory or snake for STM. You specifically mention trajectory scan in /doc too.